src/Mod/Draft/draftguitools/gui_snapper.py

```diff
@@ -553,10 +602,10 @@ def snapToExtensions(self, point, last, constrain, eline):
                         self.setCursor(tsnap[1])
                         return tsnap[2], eline
 
-        for o in (self.lastObj[1], self.lastObj[0]):
-            if o and (self.isEnabled('extension')
-                      or self.isEnabled('parallel')):
-                ob = FreeCAD.ActiveDocument.getObject(o)
+        for o in [self.lastObj[1], self.lastObj[0]]:
```

Comment:
Just nitpicking here but you'll see that in various iterations when you search for something in a list, it is theoretically faster to search in a tuple than in a list.

So, in many cases I use `for obj in (...)` rather than `for obj in [...]`. Apparently there is tiny difference.

Reply:
The way I heard is that lists are mutable (you can modify an element of the list after it has been created, while tuples not. So when creating a tuple, the exact amount of memory is allocated to it, while for lists there is more memory allocated because it might grow later on, and they are much less efficient. So the idea is always use tuples when you know your list won't need to be modified later on

Reply:
Yep, pretty much this.